File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/TVA.Core/Root/ObjectState.vb
' 05/23/2007

<Serializable()> _
Public Class ObjectState(Of TState)

#Region " Member Declaration "

    Private m_objectName As String
    Private m_currentState As TState
    Private m_previousState As TState

#End Region

#Region " Code Scope: Public "

    Public Sub New(ByVal objectName As String)

        MyClass.New(objectName, Nothing)

    End Sub

    Public Sub New(ByVal objectName As String, ByVal currentState As TState)

        MyClass.New(objectName, Nothing, currentState)

    End Sub

    Public Sub New(ByVal objectName As String, ByVal previousState As TState, ByVal currentState As TState)

        MyBase.New()
        m_objectName = objectName
        m_currentState = currentState
        m_previousState = previousState

    End Sub

    Public Property ObjectName() As String
        Get
            Return m_objectName
        End Get
        Set(ByVal value As String)
            m_objectName = value
        End Set
    End Property

    Public Property CurrentState() As TState
        Get
            Return m_currentState
        End Get
        Set(ByVal value As TState)
            m_currentState = value
        End Set
    End Property

    Public Property PreviousState() As TState
        Get
            Return m_previousState
        End Get
        Set(ByVal value As TState)
            m_previousState = value
        End Set
    End Property

#End Region

End Class

File: Source/OldCode/Archive/2005 TVA Code Library/Version/CS/TVA.Core_CS/Root/ObjectState.cs
using System.Diagnostics;
using System.Linq;
using System.Data;
using System.Collections;
using Microsoft.VisualBasic;
using System.Collections.Generic;
using System;

// 05/23/2007

namespace TVA
{
	[Serializable()]public class ObjectState<TState>
	{
		
		
		#region " Member Declaration "
		
		private string m_objectName;
		private TState m_currentState;
		private TState m_previousState;
		
		#endregion
		
		#region " Code Scope: Public "
		
		public ObjectState(string objectName) : this(objectName, null)
		{
			
			
		}
		
		public ObjectState(string objectName, TState currentState) : this(objectName, null, currentState)
		{
			
			
		}
		
		public ObjectState(string objectName, TState previousState, TState currentState)
		{
			
			m_objectName = objectName;
			m_currentState = currentState;
			m_previousState = previousState;
			
		}
		
		public string ObjectName
		{
			get
			{
				return m_objectName;
			}
			set
			{
				m_objectName = value;
			}
		}
		
		public TState CurrentState
		{
			get
			{
				return m_currentState;
			}
			set
			{
				m_currentState = value;
			}
		}
		
		public TState PreviousState
		{
			get
			{
				return m_previousState;
			}
			set
			{
				m_previousState = value;
			}
		}
		
		#endregion
		
	}
	
}
